### PR TITLE
Mesh_3: Improve Sizing_field_with_aabb_tree speed

### PR DESCRIPTION
**Pull request by lrineau**
## Summary of Changes

- Make `Sizing_field_with_aabb_tree` a lightweight object carrying a `shared_ptr`
- Use per-patch kd-trees to speed up distance queries

## Release Management

Note that `Sizing_field_with_aabb_tree` is for the moment an undocumented class from `<Mesh_3/experimental/Sizing_field_with_aabb_tree.h>`.

* Affected package(s): Mesh_3
* License and copyright ownership: maintenance by GeometryFactory, unchanged

### CODE REVIEW

File: Mesh_3/include/CGAL/Mesh_3/experimental/Sizing_field_with_aabb_tree.h [hunk 1]
```diff
@@ -71,101 +77,185 @@ struct Sizing_field_with_aabb_tree
...
+
+      struct Visit_all_primitives {
+        decltype(push_vertex) push_vertex;
+        bool go_further() const { return true; }
+        bool do_intersect(std::nullptr_t, const Node&) { return true; }
+        void intersection(std::nullptr_t, const Primitive& primitive)
+        {
+          push_vertex(primitive);
+        }
+      } visit_all_primitives{push_vertex};
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/experimental/Sizing_field_with_aabb_tree.h:139]:
```suggestion
        decltype(push_vertex) register_vertex;
        bool go_further() const { return true; }
        bool do_intersect(std::nullptr_t, const Node&) { return true; }
        void intersection(std::nullptr_t, const Primitive& primitive)
        {
          register_vertex(primitive);
        }
      } visit_all_primitives{push_vertex};
```

**Comment by sloriot** [comment]
Test failing. See [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.6-Ic-183/Polyhedron_Demo/TestReport_gimeno_Debian-testing-clang-main.gz) for example

**Comment by lrineau** [comment]
> `C:\CGAL_ROOT\CGAL-5.6-Ic-184\include\CGAL/Mesh_3/experimental/Sizing_field_with_aabb_tree.h(236): warning C4267: 'argument': conversion from 'size_t' to 'const int', possible loss of data [C:\CGAL_ROOT\CGAL-5.6-Ic-184\cmake\platforms\MSVC2015-Release-64bits\test\Polyhedron_Demo\Plugins\Mesh_3`

I pushed 8d2edc7825452fe8a6252a88dca8fa70f7cbe537 with a static cast. That will probably fix the warning.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-187

**Comment by lrineau** [comment]
TODO @lrineau: I need to re-check this branch on the `cheese` data-set.

**Comment by sloriot** [comment]
Successfully tested in 5.6-Ic-208

**Pull request merged**